### PR TITLE
Historize

### PR DESCRIPTION
**Pull request by droid42**
Added history table per entity to be able to access previous versions of the data by date.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,7 @@ go 1.13
 
 require (
 	github.com/Masterminds/squirrel v1.2.0
+	github.com/docker/go-connections v0.4.0
```
Comment (majst01) [inline go.mod:7]: This is not used

File: pkg/datastore/postgres.go [hunk 2]
```diff
@@ -353,3 +422,72 @@ func (ds *Datastore) Find(ctx context.Context, filter map[string]interface{}, re
...
+// historyTablename returns the tablename of the table-twin with historic data.
+func historyTablename(table string) string {
+	return fmt.Sprintf("%s_history", table)
+}
+
+// PbNow return
+func PbNow() *timestamp.Timestamp {
+	now, err := ptypes.TimestampProto(Now())
+	if err != nil {
+		panic(err)
```
Comment (majst01) [inline pkg/datastore/postgres.go:490]: this is a bit harsh

File: pkg/datastore/postgres.go [hunk 3]
```diff
@@ -353,3 +422,72 @@ func (ds *Datastore) Find(ctx context.Context, filter map[string]interface{}, re
...
+		cerr := rows.Close()
+		if err == nil && cerr != nil {
+			err = cerr
+		}
+	}()
+	if rows.Next() {
+		return rows.Scan(ve)
+	}
+	// we have no row
+	return NewNotFoundError(fmt.Sprintf("entity of type:%s with id:%s not found", jsonField, id))
```
Comment (majst01) [inline pkg/datastore/postgres.go:461]: Maybe adding created_at here is helpful

File: pkg/gen/genscanvaluer.go [hunk 4]
```diff
@@ -143,6 +143,14 @@ const schemaTemplate = `
 		{{ .typeNameLower }} JSONB NOT NULL
 	);
 	CREATE INDEX IF NOT EXISTS {{ .typeNameLower }}_idx ON {{ .tableName }} USING GIN({{ .typeNameLower }});
+
+	CREATE TABLE IF NOT EXISTS {{ .tableName }}_history (
+		id         text NOT NULL,
+		op		   char NOT NULL,
```
Comment (majst01) [inline pkg/gen/genscanvaluer.go:149]: mixed tabs/spaces

File: pkg/datastore/postgres.go (hunk 2 above)
Comment (droid42) [reply pkg/datastore/postgres.go:490]: The only possibility for this error to occur is in testing when someone fails to provide a correct Now-func when overriding for testing at a defined point in time.
What is more: it is exactly what github.com/golang/protobuf@v1.3.5/ptypes/timestamp.go:103 does.
So I would suggest to leave it this way, even I also don't like panics in "harmless" funcs.

File: pkg/datastore/postgres.go (hunk 3 above)
Comment (droid42) [reply pkg/datastore/postgres.go:461]: :+1:

File: pkg/gen/genscanvaluer.go (hunk 4 above)
Comment (droid42) [reply pkg/gen/genscanvaluer.go:149]: Ok

File: go.mod (hunk 1 above)
Comment (droid42) [reply go.mod:7]: Ok

File: pkg/datastore/postgres.go (hunk 2 above)
Comment (majst01) [reply pkg/datastore/postgres.go:490]: Decided this will not happen

**Pull request merged**